akka-actor-typed-tests/src/test/scala/docs/akka/typed/StyleGuideDocExamples.scala

```diff
@@ -522,6 +523,113 @@ object StyleGuideDocExamples {
+
+    sealed trait Command
+    case object Down extends Command
+    final case class GetValue(replyTo: ActorRef[Value]) extends Command
+    final case class Value(n: Int)
+
+    //#get-handler-partial
+    def getHandler(value: Int): PartialFunction[Command, Behavior[Command]] = {
+      case GetValue(replyTo) =>
+        replyTo ! Value(value)
+        Behaviors.same
+    }
+    //#get-handler-partial
+
+    //#set-handler-non-zero-partial
+    def setHandlerNotZero(value: Int): PartialFunction[Command, Behavior[Command]] = {
+      case Down =>
+        if (value == 1)
+          zero
+        else
+          nonZero(value - 1)
+    }
+    //#set-handler-non-zero-partial
+
+    //#handle-partial
+    def handle(command: Command, handlers: List[PartialFunction[Command, Behavior[Command]]]): Behavior[Command] = {
+      handlers match {
+        case Nil          => Behaviors.unhandled
+        case head :: tail => head.applyOrElse(command, handle(_, tail))
+      }
+    }
+    //#handle-partial
+
+    //#top-level-behaviors-partial
+    val zero: Behavior[Command] = Behaviors.receiveMessage { command =>
+      handle(command, List(getHandler(0)))
+    }
+
+    def nonZero(initialCapacity: Int): Behavior[Command] = Behaviors.receiveMessage { command =>
+      handle(command, List(getHandler(initialCapacity), setHandlerNotZero(initialCapacity)))
+    }
+    //#top-level-behaviors-partial
+  }
+
+  object BehaviorCompositionWithReceivePartial {
+
+    //#messages-sealed-composition
+    sealed trait Command
+    case object Down extends Command
+    final case class GetValue(replyTo: ActorRef[Value]) extends Command
+    final case class Value(n: Int)
+    //#messages-sealed-composition
+
+    //#get-handler
+    def getHandler(value: Int): Behavior[Command] = Behaviors.receiveMessagePartial {
+      case GetValue(replyTo) =>
+        replyTo ! Value(value)
+        Behaviors.same
+    }
+    //#get-handler
+
+    //#set-handler-non-zero
+    def setHandlerNotZero(value: Int): Behavior[Command] = Behaviors.receiveMessagePartial {
+      case Down =>
+        if (value == 1)
+          zero
+        else
+          nonZero(value - 1)
+    }
+    //#set-handler-non-zero
+
+    //#set-handler-zero
+    val setHandlerZero: Behavior[Command] = Behaviors.receivePartial {
+      case (context, Down) =>
+        context.log.error("Counter is already at zero!")
+        Behaviors.same
+    }
+    //#set-handler-zero
+
+    //#handle
+    @tailrec def handle(
+        command: Command,
+        handlers: List[Behavior[Command]],
+        ctx: ActorContext[Command]): Behavior[Command] = {
+      handlers match {
+        case Nil          => Behaviors.unhandled
+        case head :: tail =>
+          //Process the given message with the next Behavior in the list
+          val next = Behavior.interpretMessage(head, ctx, command)
+          if (Behavior.isUnhandled(next)) handle(command, tail, ctx)
+          else next
+      }
```

Comment:
This is safe but I'm not quite convinced this kind of low level calling into the interpreter belongs in the style guide (even if it _is_ public API). I think I'd prefer that the composition was done using PartialFunction.orElse and avoid this.

Reply:
I can reverse the order and fist explain plain function composition and then Partial Behavior composition.

But I think the Behavior composition should be explained somewhere IMO as it gives a better understanding of why `Behaviors.receivePartial` exists and can be used for.

Reply:
Do you think this example using partial Behaviors should be removed completely?

Reply:
Yes, I think so, composing the actual behavior is a slipperly slope to end up with the original `.orElse` problem, even if this sample explicitly covers a single top message and shares actor context type I'm pretty sure it will inspire more "clever/advanced" usage to try to compose heterogenous protocols. Better not lead users down that path of own-foot-shooting.

Reply:
For the reference there are more details about the problems with composing heterogenous protocols here: https://github.com/akka/akka/issues/26867

Reply:
And a warning box and a link to the Github issue to warn people?

Reply:
I think just not encouraging going there is enough. Given that it is a style guide I think it should show good idiomatic solutions/patterns for common problems, not cover/warn about all possible bad ideas.

Reply:
I will remove this example then and simplify the `handle` method then.